### PR TITLE
hood: recursive `|tomb`

### PR DESCRIPTION
**Pull request by ryjm**
`|tomb-lobe` uses %lobe which ended up being much faster.

### CODE REVIEW

**Comment by zalberico** [comment]
@ryjm can you clarify what this is actually for or write up and link it to an issue about what problem this is solving? As is it's hard for someone to evaluate and approve/merge.

**Comment by ryjm** [comment]
> @ryjm can you clarify what this is actually for or write up and link it to an issue about what problem this is solving? As is it's hard for someone to evaluate and approve/merge.

pairs with #6057 somewhat - sometimes you want to remove/tombstone every file in a directory. 

both of these PRs are rather small changes and only touch generators, so they're mostly QoL stuff.

**Comment by zalberico** [comment]
@ryjm thanks - @belisarius222, @philipcmonk adding you to this for review for approval/merge (generally I'm trying to get things that can merge to merge so we don't have PRs open unnecessarily).

**Comment by zalberico** [comment]
@belisarius222 ping on this one, trying to keep some of these that look ready moving along

**Comment by belisarius222** [comment]
@ryjm Can you please explain the intended behavior of this generator?  Then I'll review it.

**Comment by jalehman** [comment]
@ryjm Please comment.

**Comment by ryjm** [comment]
Currently you can only tombstone single files and not dirs - I made the original `|tomb` recursive, and added a new generator that's also recursive but much faster, using `%lobe`.  PR'd both just to cover all bases. 

Happy to update to coalesce into one generator, just was hoping to get some feedback on which strategy is preferable, why using `%lobe` is so much faster, etc.

File: pkg/arvo/gen/hood/tomb.hoon
Comment (philipcmonk) [inline pkg/arvo/gen/hood/tomb.hoon:19]: Indentation

File: pkg/arvo/gen/hood/tomb.hoon
Comment (philipcmonk) [inline pkg/arvo/gen/hood/tomb.hoon:57]: Indentation

**Review by philipcmonk — CHANGES_REQUESTED** [review]
I think this makes sense, just some formatting issues

File: pkg/arvo/gen/hood/tomb-lobe.hoon
Comment (philipcmonk) [inline pkg/arvo/gen/hood/tomb-lobe.hoon:11]: very misleading indentation

Comment (ryjm) [reply pkg/arvo/gen/hood/tomb-lobe.hoon:11]: oof yeah this one is terrible

**Comment by ryjm** [comment]
i don't have write access anymore, should i make a PR? here's a patch if someone wants to just apply it instead.

```
From af17b3ad52b5aa11e4b86009b2c1aecbf9c7bf2b Mon Sep 17 00:00:00 2001
From: ryjm <jraydermiller@gmail.com>
Date: Mon, 6 Feb 2023 13:51:29 -0500
Subject: [PATCH] fix indentation

---
 pkg/arvo/gen/hood/tomb-lobe.hoon |  2 +-
 pkg/arvo/gen/hood/tomb.hoon      | 24 ++++++++++++------------
 2 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/pkg/arvo/gen/hood/tomb-lobe.hoon b/pkg/arvo/gen/hood/tomb-lobe.hoon
index 6668ed4966..4987c283e2 100644
--- a/pkg/arvo/gen/hood/tomb-lobe.hoon
+++ b/pkg/arvo/gen/hood/tomb-lobe.hoon
@@ -8,7 +8,7 @@
 |=  [[now=@da eny=@uvJ bec=beak] [target=path ~] dry=_|]
 :-  %helm-pans
 =;  cards
-?.  dry  cards
+  ?.  dry  cards
   %-  (slog leaf+"card: {<cards>}" ~)  ~
 =|  lubs=(list note-arvo)
 |-  ^-  (list note-arvo)
diff --git a/pkg/arvo/gen/hood/tomb.hoon b/pkg/arvo/gen/hood/tomb.hoon
index 5db418fa9c..8f77e0377f 100644
--- a/pkg/arvo/gen/hood/tomb.hoon
+++ b/pkg/arvo/gen/hood/tomb.hoon
@@ -54,18 +54,18 @@
   --
 ::
 ++  lobes
-=|  lubs=(list lobe)
-|-  ^-  (list lobe)
-=+  .^(=arch %cy target)
-?~  fil.arch
-  =/  dirs  ~(tap by dir.arch)
-  %-  zing
-  %+  turn  dirs
-  |=  [kid=@ta ~]
-  =/  paf=path  /[kid]
-  =/  kud=path  `path`(weld target /[kid])
-  ^$(target kud)
- (snoc lubs u.fil.arch)
+  =|  lubs=(list lobe)
+  |-  ^-  (list lobe)
+  =+  .^(=arch %cy target)
+  ?~  fil.arch
+    =/  dirs  ~(tap by dir.arch)
+    %-  zing
+    %+  turn  dirs
+    |=  [kid=@ta ~]
+      =/  paf=path  /[kid]
+      =/  kud=path  `path`(weld target /[kid])
+      ^$(target kud)
+  (snoc lubs u.fil.arch)
 ::
 ++  notes
   |=  norms=(set [ship desk tako norm path])
-- 
2.37.3

```

**Comment by jalehman** [comment]
@mcevoypeter Any idea why the mingw build is running on this branch?

**Comment by mcevoypeter** [comment]
> Any idea why the mingw build is running on this branch?

Which job are you referring to? The jobs I see under the `Checks` tab of this PR both show it only running on `ubuntu-latest`.

**Comment by philipcmonk** [comment]
The branch was out of date, Jake just merged develop into it

**Comment by ryjm** [comment]
@philipcmonk @belisarius222 

i added a fallback to `|rm` option and fixed the recursion (wasn't working on files/dirs sharing the same name). probably needs another review if you don't mind.

File: pkg/arvo/gen/hood/tomb.hoon
Comment (belisarius222) [inline pkg/arvo/gen/hood/tomb.hoon:23]: I don't think you need this `doms` local variable, or the `|-` below it.  You should be able to use the `|=  lob=lobe` gate as the outer recursion point.

Comment (belisarius222) [reply pkg/arvo/gen/hood/tomb.hoon:23]: nvm I'm wrong

**Review by belisarius222 — APPROVED** [review]

**Comment by belisarius222** [comment]
CI is borked.  Merging.

**Pull request merged**
